src/cookies.js

```diff
@@ -84,9 +84,25 @@ export function setCookie(win, name, value, expirationTime, opt_options) {
  * @param {string|undefined} domain
  */
 function trySetCookie(win, name, value, expirationTime, domain) {
+  // We do not allow setting cookies on the domain that contains both
+  // the cdn. and www. hosts.
+  if (domain == 'ampproject.org') {
+    // Actively delete them.
+    win.document.cookie = encodeURIComponent(name) + '=delete' +
```

Comment:
Instead of duplicating all this code, why not just set `value` and `expirationTime`?